kratos/spatial_containers/search_wrapper.h

```diff
@@ -0,0 +1,1024 @@
+            mSettings(Settings)
+    {
+        // Validate and assign defaults
+        mSettings.ValidateAndAssignDefaults(GetDefaultParameters());
+
+        // Create base search object
+        if constexpr (IsGeometricalObjectBins) {
+            mpSearchObject = Kratos::make_shared<TSearchObject>(rGeometricalObjectsVector.begin(), rGeometricalObjectsVector.end());
+        } else { // Otherwise we assume it will be some kind of spatial search
+            // Prepare the point vector if geometrical objects are provided
+            if (rGeometricalObjectsVector.size() > 0) {
+                // Defining the PointVector
+                mpPointVector = Kratos::make_unique<PointVector>(SearchUtilities::PreparePointsSearch(rGeometricalObjectsVector));
+
+                // Create the search object
+                if constexpr (!IsDynamicBins) {
+                    const int bucket_size = mSettings["bucket_size"].GetInt();
+                    mpSearchObject = Kratos::make_shared<TSearchObject>(mpPointVector->begin(), mpPointVector->end(), bucket_size);
+                } else {
+                    mpSearchObject = Kratos::make_shared<TSearchObject>(mpPointVector->begin(), mpPointVector->end());
+                }
+            }
+        }
+
+    #ifdef KRATOS_USING_MPI
+        // Set up the global bounding boxes
+        InitializeGlobalBoundingBoxes();
+    #endif
+    }
+
+    /// Destructor.
+    ~SearchWrapper() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Getting the bins bounding box
+     * @return The bounding box of the bins
+     */
+    BoundingBox<Point> GetBoundingBox() const;
+
+    /**
+     * @brief This method takes a point and finds all of the objects in the given radius to it.
+     * @details The result contains the object and also its distance to the point.
+     * @param itPointBegin The first point iterator
+     * @param itPointEnd The last point iterator
+     * @param Radius The radius to be checked
+     * @param rResults The results of the search
+     * @param ClearSolution Clear the current solution
+     * @tparam TPointIteratorType The type of the point iterator
+     */
+    template<class TPointIteratorType>
+    void SearchInRadius(
+        TPointIteratorType itPointBegin,
+        TPointIteratorType itPointEnd,
+        const double Radius,
+        ResultContainerVectorType& rResults,
+        const bool ClearSolution = true
+        )
+    {
+#ifdef KRATOS_USING_MPI
+    // Call distributed version
+    if (mrDataCommunicator.IsDistributed()) {
+        DistributedSearchInRadius(itPointBegin, itPointEnd, Radius, rResults, ClearSolution);
+    } else { // Call serial version
+        SerialSearchInRadius(itPointBegin, itPointEnd, Radius, rResults, ClearSolution);
+    }
+#else // Calling just serial method
+    SerialSearchInRadius(itPointBegin, itPointEnd, Radius, rResults, ClearSolution);
+#endif
```

Comment:
For the versions where you call searches over all points there is no need to use `#ifdefs,` the overhead of checking if distributed or not even in serial is negligible (but maintain if there are function that could be called inside a loop on individual points, there is important to do this)..

```suggestion
    if (mrDataCommunicator.IsDistributed()) {
        DistributedSearchInRadius(itPointBegin, itPointEnd, Radius, rResults, ClearSolution);
    } else { // Call serial version
        SerialSearchInRadius(itPointBegin, itPointEnd, Radius, rResults, ClearSolution);
    }
```

Reply:
Okay